### PR TITLE
Add Credo analyzer

### PR DESCRIPTION
**Pull request by noteven**
To help keep a consistent and hygienic code style, follow good practices, and avoid some known pitfalls it would be good to add Credo.

As such I've taken the liberty of adding in the dependency as well as have enable some of the non-default opinionated checks.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -11,21 +11,20 @@ defmodule Silicon.MixProject do
...
   end
 
-  # Run "mix help deps" to learn about dependencies.
   defp deps do
     [
       {:rustler, "~> 0.30.0", runtime: false},
-      {:ex_doc, "~> 0.31", only: :dev, runtime: false},
+
+      # Developer tooling
       {:type_check, "~> 0.13.3"},
```
Comment (AviKav) [inline mix.exs]: `type_check` shouldn't be under dev tooling

Comment (noteven) [reply mix.exs]: Isn't it though? It doesn't provide any end-user features as far as I am aware?

Comment (AviKav) [reply mix.exs]: Currently, it's only used for runtime validation since rustler's suck

File: .credo.exs [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      #
+      # If you create your own checks, you must specify the source files for
+      # them here, so they can be loaded by Credo before running the analysis.
+      #
+      requires: [],
+      #
+      # If you want to enforce a style guide and need a more traditional linting
+      # experience, you can change `strict` to `true` below:
+      #
+      strict: false,
```
Comment (ryanwinchester) [inline .credo.exs]:
```suggestion
      strict: true,
```

if you're going in, go hard

File: mix.exs (hunk 1 above)
Comment (noteven) [reply mix.exs]: Ah so it is used at runtime and not only during development like `mix dialyzer`? If so then I'll move it as recommended.

**Pull request merged**